tests/system/responses/test_exception_async_rtu_responses.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+import struct
+from functools import partial
+
+from ..validators import validate_response_mbap
+
+from umodbus.client.serial import rtu
+from umodbus.client.serial.redundancy_check import (get_crc, validate_crc,
+                                                    add_crc, CRCError)
+
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.mark.parametrize('function_code, quantity', [
+    (1, 0),
+    (2, 0),
+    (3, 0),
+    (4, 0),
+    (1, 0x07D0 + 1),
+    (2, 0x07D0 + 1),
+    (3, 0x007D + 1),
+    (4, 0x007D + 1),
+])
+async def test_request_returning_invalid_data_value_error(
+    rtu_server, async_serial_streams, function_code, quantity):
+    """ Validate response PDU of request returning excepetion response with
+    error code 3.
+    """
+    starting_address = 0
+    slave_id = 1
+    adu = add_crc(struct.pack('>BBHH', slave_id, function_code,
+                              starting_address, quantity))
+
+    reader, writer = async_serial_streams
+    writer.write(adu)
+    await writer.drain()
+    resp = await reader.read(rtu_server.serial_port.in_waiting)
+
+    validate_crc(resp)
+    assert struct.unpack('>BB', resp[1:-2]) == (0x80 + function_code, 3)
```

Comment:
Referring back to a previous comment, line 35-41 are pretty similar to those in other tests, could you factor them  out into a utility?

Reply:
Sure, it should be better now.